weld/subsystem/src/main/java/org/jboss/as/weld/WeldProvider.java

```diff
@@ -58,6 +59,9 @@ static void containerShutDown(Container container) {
 
     @Override
     public CDI<Object> getCDI() {
+        if (ModuleGroupSingletonProvider.deploymentClassLoaders.isEmpty()) {
+            throw new IllegalStateException();
```

Comment:
Will this be caught and dealt with up the call stack? My main concern is just reporting to the user what the issue actually is.

Reply:
No, this is how the contract is defined: https://github.com/cdi-spec/cdi/blob/master/api/src/main/java/javax/enterprise/inject/spi/CDIProvider.java#L34. So it's up to the `javax.enterprise.inject.spi.CDI` client. See also https://github.com/cdi-spec/cdi/blob/master/api/src/main/java/javax/enterprise/inject/spi/CDI.java#L60.